codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AssembleMiddlewareStack.java

```diff
@@ -41,62 +48,100 @@ public byte getOrder() {
                         .registerMiddleware(MiddlewareRegistrar.builder()
                                 .resolvedFunction(SymbolUtils.createValueSymbolBuilder(
                                         "AddResolveServiceEndpointMiddleware", AwsGoDependency.AWS_MIDDLEWARE)
-                                .build())
+                                        .build())
                                 .useClientOptions()
                                 .build())
                         .build(),
 
-                // Add attemptClockSkew middleware to operation stack
+                // Add unsigned payload middleware to operation stack
                 RuntimeClientPlugin.builder()
+                        .operationPredicate((model, service, operation) -> {
+                            boolean hasSigV4Auth = hasSigV4AuthScheme(model, service, operation);
```

Comment:
nit: This assignment isn't necessary

Reply:
mainly just to make the lines shorter, but can drop extra variable. :)